pkg/image/validation/kubernetes.go

```diff
@@ -143,69 +142,115 @@ func validateManifestURLs(k8s *image.Kubernetes) []FailedValidation {
 	return failures
 }
 
-func validateHelmCharts(k8s *image.Kubernetes, imageConfigDir string) []FailedValidation {
+func validateHelm(k8s *image.Kubernetes, imageConfigDir string) []FailedValidation {
 	var failures []FailedValidation
 
-	if len(k8s.HelmCharts) == 0 {
+	if len(k8s.Helm.Charts) == 0 {
+		return failures
+	}
+
+	if len(k8s.Helm.Repositories) == 0 {
+		failures = append(failures, FailedValidation{
+			UserMessage: "Helm Charts defined with no Helm Repository defined.",
+		})
+
 		return failures
 	}
 
 	seenHelmCharts := make(map[string]bool)
-	for _, chart := range k8s.HelmCharts {
-		if chart.Name == "" {
-			failures = append(failures, FailedValidation{
-				UserMessage: "Helm Chart 'name' field must be defined.",
-			})
-		}
+	seenHelmRepos := make(map[string]bool)
+	for _, chart := range k8s.Helm.Charts {
+		c := chart
+		failures = append(failures, validateChart(&c, seenHelmCharts, imageConfigDir)...)
 
-		if chart.Repo == "" {
-			failures = append(failures, FailedValidation{
-				UserMessage: "Helm Chart 'repo' field must be defined.",
-			})
-		} else if !strings.HasPrefix(chart.Repo, "http") && !strings.HasPrefix(chart.Repo, "oci://") {
-			failures = append(failures, FailedValidation{
-				UserMessage: "Helm Chart 'repo' field must begin with either 'oci://', 'http://', or 'https://'.",
-			})
-		}
+		seenHelmCharts[chart.Name] = true
+		seenHelmRepos[chart.RepositoryName] = true
```

Comment:
I don't think this will work as expected since the `seenHelmCharts` is only appended once per iteration. Even if it's _fine_ I'd still suggest to extract the duplication catch out of this loop for simplicity.